providers/com/peerj/harvester.py

```diff
@@ -0,0 +1,56 @@
+import arrow
+from furl import furl
+
+import logging
+from share import Harvester
+
+logger = logging.getLogger(__name__)
+
+
+class PeerJHarvester(Harvester):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.base_url = 'https://peerj.com/articles/index.json'
+        self.base_preprint_url = 'https://peerj.com/preprints/index.json'
+
+    def do_harvest(self, start_date: arrow.Arrow, end_date: arrow.Arrow):
+
+        return self.fetch_records(self.base_url, start_date, end_date)
+
+    def fetch_records(self, url, start_date, end_date):
+
+        logger.info('Making request to {}'.format(url))
+        for url in self.build_url():
+            if self.base_preprint_url in url:
+                yield from self.fetch_page(url, start_date, end_date, 'preprint-')
+            yield from self.fetch_page(url, start_date, end_date)
+
+        logger.info("PeerJ has been harvested")
+
+    def fetch_page(self,url, start_date, end_date, preprint=''):
+        records = self.requests.get(url).json()
+        next_page = records['_links']
+        for record in records['_items']:
+            if arrow.get(record['date']) < start_date:
+                return
+
+            if arrow.get(record['date']) > end_date:
+                continue
+
+            yield (preprint+record['identifiers']['peerj'], record)
```

Comment:
Yeah, that's why I put that in there. Sometimes they will both have the number 89 or something so I just appended 'preprints-' to all the ones resulting from the preprints journal. CS already has the append in front of it.